### ISSUE TITLE
syntax-rules renaming error?

### ISSUE THREAD

**Issue by stibear** [picrin-scheme/picrin#229]
https://gist.github.com/stibear/96701923b644203acd97

expected:
```scheme
> (mbi-dirty-v1 10 (+ i 1))
11
```
but actually:
```scheme
> (mbi-dirty-v1 10 (+ i 1))
error: unbound variable i@4219
```

**Comment by wasabiz** [comment]
Hi, we've replaced macro expander with the new improved version and rewritten syntax-rules expander from scratch, yay! (See [Picrin:マクロシステム](http://practical-scheme.net/wiliki/wiliki.cgi?Picrin%3Aマクロシステム) for details). So, I tested the code by stibear this night but got an error while expanding syntax-rulesreading 'error: "cannot expand improper list: (cont-head symb-l . cont-args)"'. Since the original author of picrin's syntax-rules is @zeptometer, I asked him to check and fix the bug via oral conversation. He is so 優秀 as to not need weeks to fix this bug. I think we can close this issue soon. (Probably the original problem about hygienic macro is already solved by the new expansion algorithm. The problem we have now is simply a bug of analysis of pattern rule or synthesis of expander.)
